logging/log.ts

```diff
@@ -75,29 +75,29 @@ const colorMap = {
 };
 
 export const coloredStringLogger: LogFn = (msg, _ctx, level) => {
-  const color = colorMap[level];
+  const color = colorMap[level ?? 'info'];
   console.log(`[river:${color}${level}\u001b[0m] ${msg}`);
 };
 
 export const jsonLogger: LogFn = (msg, ctx, level) => {
   console.log(JSON.stringify({ msg, ctx, level }));
 };
 
-export let log: BaseLogger | undefined = undefined;
+export let log: Logger | undefined = undefined;
 export function bindLogger(
-  fn: LogFn | BaseLogger | undefined,
+  fn: LogFn | Logger | undefined,
   level?: LoggingLevel,
-) {
+): Logger | undefined {
   if (!fn) {
     log = undefined;
     return;
   }
 
-  if (fn instanceof BaseLogger) {
-    log = fn;
-    return fn;
+  if (fn instanceof Function) {
```

Comment:
```suggestion
  if (typeof fn === 'function') {
```

It's generally considered a safer choice and wont have issues with cross context functions.